### PR TITLE
Move some native assemblies to Tooling for System.Net.Http

### PR DESCRIPTION
**Pull request by BrennanConroy**
@moozzyk @davidfowl

### CODE REVIEW

File: makefile.shade [hunk 1]
```diff
@@ -516,6 +516,13 @@ var CAN_BUILD_ONECORE = '${Directory.Exists(VS_ONECORE_ROOT) && Directory.GetFil
             "Microsoft.Dnx.Project",
             "Microsoft.Dnx.Tooling"
         };
+        
+        // native assemblies loaded by System.Net.Http.dll at runtime
+        var httpNativeAssemblies = new []
```
Comment (davidfowl) [inline makefile.shade]: Call it native libraries

File: makefile.shade [hunk 2]
```diff
@@ -611,6 +618,18 @@ var CAN_BUILD_ONECORE = '${Directory.Exists(VS_ONECORE_ROOT) && Directory.GetFil
                          || (Path.GetFileName(f).StartsWith("System") && Path.GetFileName(f).EndsWith(nativeExtension))))
                 {
                     var targetFilePath = Path.Combine(target.TargetFolder, Path.GetFileName(file));
+                    if (file.EndsWith(nativeExtension))
+                    {
+                        foreach (var native in httpNativeAssemblies)
```
Comment (moozzyk) [inline makefile.shade]: Why not just blindly copy the files from httpNativeAssemblies instead of checking if the names match for each so/dylib?

**Comment by davidfowl** [comment]
:shipit: When the build passes.

File: makefile.shade [hunk 3]
```diff
@@ -614,6 +621,13 @@ var CAN_BUILD_ONECORE = '${Directory.Exists(VS_ONECORE_ROOT) && Directory.GetFil
                     Log.Info("Copying " + file + " to " + targetFilePath);
                     File.Copy(file, targetFilePath, true);
                 }
+                
+                foreach (var native in nativeLibraries)
+                {
+                    var targetFilePath = Path.Combine(target.TargetFolder, "lib", "Microsoft.Dnx.Tooling", native + nativeExtension);
+                    Log.Info("Moving " + native + " to " + targetFilePath);
+                    File.Move(Path.Combine(target.TargetFolder, native + nativeExtension), targetFilePath);
```
Comment (davidfowl) [inline makefile.shade]: Make sure this only happens for core clr only.

File: makefile.shade [hunk 4]
```diff
@@ -614,6 +621,20 @@ var CAN_BUILD_ONECORE = '${Directory.Exists(VS_ONECORE_ROOT) && Directory.GetFil
...
+                if (target.OS != "win")
+                {
+                    foreach (var native in nativeLibraries)
+                    {
+                        var targetFilePath = Path.Combine(target.TargetFolder, "lib", "Microsoft.Dnx.Tooling", native + nativeExtension);
+                        var sourceFile = Path.Combine(target.TargetFolder, native + nativeExtension);
+                        if (File.Exists(sourceFile))
+                        {
+                            Log.Info("Moving " + native + " to " + targetFilePath);
+                            File.Move(Path.Combine(target.TargetFolder, native + nativeExtension), targetFilePath);
```
Comment (moozzyk) [inline makefile.shade]: @BrennanConroy - did you want to use sourceFile in the File.Move? - from what I can see the first arg has the same value...

Comment (BrennanConroy) [inline makefile.shade]: Ya :/

Comment (davidfowl) [inline makefile.shade]: Did you test this @BrennanConroy ?

Comment (BrennanConroy) [inline makefile.shade]: Yep, there isn't a problem, just can replace some of the stuff with the variable to make it look nicer.

**Comment by davidfowl** [comment]
@BrennanConroy close your bugs https://github.com/aspnet/dnx/issues/2607 :smile:

**Comment by BrennanConroy** [comment]
Troll...

File: makefile.shade (hunk 4 above)
Comment (moozzyk) [inline makefile.shade]: One more (after)thought - shouldn't we fail (or at least warn) if the expected *Native.{so|dylib} - does not exist? If it does not exist it means that either CoreClr will be busted or it is no longer needed and the script should be updated accordingly.

**Pull request merged**